plotly/tools.py

```diff
@@ -1269,3 +1277,628 @@ def return_figure_from_figure_or_data(figure_or_data, validate_figure):
+            raise exceptions.PlotlyError("Your data array could not be"
+                                         "flattened! Make sure x, y, u, and v"
+                                         "are lists or numpy ndarrays!")
+
+    def validate(self):
+        """
+        Validates that args and kwargs meet criteria,
+        specifically that scale and arrow_scale are positive
+        and that x, y, u, and v are the same length
+
+        :raises: (ValueError) If scale or arrow_scale is < 1.
+        :raises: (PlotlyError) If x, y, u, and v are not the same length.
+        """
+        if self.scale <= 0:
+            raise ValueError("scale must be > 0")
+        if self.arrow_scale <= 0:
+            raise ValueError("arrow_scale must be > 0")
+        if (len(self.x) != len(self.y) or len(self.u) != len(self.v) or
+           len(self.x) != len(self.u)):
+            raise exceptions.PlotlyError("x, y, u, and v should all be the"
+                                         " same length (or size if ndarray)")
+
+    def scale_uv(self):
+        """
+        Scales u and v. u and v are added to x and y to get the
+        endpoints of the arros so a smaller scale value will
+        result in less overlap of arrows.
+        """
+        self.u = [i * self.scale for i in self.u]
+        self.v = [i * self.scale for i in self.v]
+
+    def get_barbs(self):
+        """
+        Gets endpoint of the barb and then zips startpoint and endpoint
+        pairs to create 2 lists: x_values for barbs and y values for barbs
+
+        :rtype: (list) barb_x: list of startpoint and endpoint x_value
+            pairs separated by a None to create the barb of the arrow.
+        :rtype: (list) barb_y: list of startpoint and endpoint y_value
+            pairs separated by a None to create the barb of the arrow.
+        """
+        self.end_x = [i + j for i, j in zip(self.x, self.u)]
+        self.end_y = [i + j for i, j in zip(self.y, self.v)]
+        empty = [None] * len(self.x)
+        barb_x = self.flatten(zip(self.x, self.end_x, empty))
+        barb_y = self.flatten(zip(self.y, self.end_y, empty))
+        return barb_x, barb_y
+
+    def get_quiver_arrows(self):
+        """
+        Gets length of each barb then calculates the length of each side of
+        the arrow. Gets angle of barb and applies angle (kwarg) to each
+        side of the arrowhead. Next uses arrow_scale to scale the length of
+        arrowhead and creates x and y values for arrowhead point1 and point2.
+        Finally x and y values for point1, endpoint and point2s for each
+        arrowhead are separated by a None and zipped to create lists of x and
+        y values for the arrows.
+
+        :rtype: (list) arrow_x: list of point1, endpoint, point2 x_values
+            separated by a None to create the arrowhead.
+        :rtype: (list) arrow_y: list of point1, endpoint, point2 y_values
+            separated by a None to create the barb of the arrow.
```

Comment:
:cow2: Hmm, not positive on this one, but I'm *pretty* sure only one rtype can be given as you *must* return one and only one value from a function.

So, the convention would be:

```
"""
:rtype: (list, list)
"""
```